services/src/main/java/org/keycloak/services/managers/RealmManager.java

```diff
@@ -719,7 +720,15 @@ public void setupClientServiceAccountsAndAuthorizationOnImport(RealmRepresentati
             ClientManager clientManager = new ClientManager(this);
 
             for (ClientRepresentation client : clients) {
-                ClientModel clientModel = this.getRealmByName(rep.getRealm()).getClientById(client.getId());
+                RealmModel realmModel = this.getRealmByName(rep.getRealm());
+
+                ClientModel clientModel = Optional.ofNullable(client.getId())
+                        .map(realmModel::getClientById)
+                        .orElse(realmModel.getClientByClientId(client.getClientId()));
```

Comment:
Prefer `orElseGet`, otherwise the lookup by `getClientByClientId` is always performed, even when client can be found by id.

Reply:
@hmlnarik You're right. Thanks!